src/libexpr/primops.cc

```diff
@@ -2928,6 +2928,57 @@ static RegisterPrimOp primop_partition({
     .fun = prim_partition,
 });
 
+static void prim_groupBy(EvalState & state, const Pos & pos, Value * * args, Value & v)
+{
+    state.forceFunction(*args[0], pos);
+    state.forceList(*args[1], pos);
+
+    ValueVectorMap attrs;
+
+    for (unsigned int n = 0; n < args[1]->listSize(); ++n) {
```

Comment:
This can be written using the list iterator now, i.e. `for (auto vElem : args[1]->listItems()) ...`.

Reply:
Nice, did this